### DISCUSSION TITLE
components subfolder declared in dirs not recognized

### DISCUSSION THREAD

**Discussion by Yegaston** [nuxt/nuxt#17342] [Questions] [answered]
Hi!

I struggling with the following problem

When i declare a sub folder inside of components in nuxt.config.ts, then when a want to use in a page the component inside that subfolder not works and throw a error in the terminal

Example:

```
folder structure
components -> shared -> Typography
```

```
nuxt.config.ts code

export default defineNuxtConfig({
 ...
  // components
  components: {
    global: true,
    dirs: [
      '~/components',
      '~/components/shared',
    ]
  },
})

```

```
usage

<template>
  <Typography />
</template>

```

This throw this error
```
[Vue warn]: Failed to resolve component: MenuHeader
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement.
```

This flow doesn't works. But in the beta works fine. Anybody knows is any way of replicate that way of imports components??

**Comment by Flanker72** [comment, marked as answer]
You have nested dirs defined. Nuxt already found the component in the 1st folder and generated `SharedTypography` component definition. You can open `.nuxt/components.d.ts` file and check what components are defined.

To reach the desired result you may try to move `components/shared` folder to `shared` and specify it in the `dirs` as now.

**Reply by Yegaston** [reply]
Thanks!!
